packages/extension-ui/src/components/Address.tsx

```diff
@@ -199,59 +194,62 @@ function Address({
-            {parentName ? (
-              <>
-                <div className='name-banner'>
-                  <FontAwesomeIcon
-                    className='deriveIcon'
-                    icon={faCodeBranch}
+        <div className='info'>
+          {parentName ? (
+            <>
+              <div className='name-banner'>
+                <FontAwesomeIcon
+                  className='deriveIcon'
+                  icon={faCodeBranch}
+                />
+                <div
+                  className='parentName'
+                  data-field='parent'
+                  title={parentNameSuri}
+                >
+                  <Svg
+                    className='subaccount-icon'
+                    src={subAccountIcon}
                   />
-                  <div
-                    className='parentName'
-                    data-field='parent'
-                    title={parentNameSuri}
-                  >
-                    <Svg
-                      className='subaccount-icon'
-                      src={subAccountIcon}
-                    />
-                    <span>{parentNameSuri}</span>
-                  </div>
-                </div>
-                <div className='name displaced'>
-                  <Name />
+                  <span>{parentNameSuri}</span>
                 </div>
-              </>
-            ) : (
-              <div
-                className='name'
-                data-field='name'
-              >
-                <Name />
               </div>
-            )}
-            {chain?.genesisHash && chain?.genesisHash !== ALEPH_ZERO_GENESIS_HASH && (
-              <div
-                className='banner chain'
-                data-field='chain'
-                style={chain.definition.color ? { backgroundColor: chain.definition.color } : undefined}
-              >
-                {chain.name.replace(' Relay Chain', '')}
+              <div className='name displaced'>
+                <Name />
               </div>
-            )}
-            <div className='addressDisplay'>
+            </>
+          ) : (
+            <div
+              className='name'
+              data-field='name'
+            >
+              <Name />
+            </div>
+          )}
+          {chain?.genesisHash && chain?.genesisHash !== ALEPH_ZERO_GENESIS_HASH && (
+            <div
+              className='banner chain'
+              data-field='chain'
+              style={chain.definition.color ? { backgroundColor: chain.definition.color } : undefined}
+            >
+              {chain.name.replace(' Relay Chain', '')}
+            </div>
+          )}
+          <div className='addressDisplay'>
+            <CopyToClipboard
+              onCopy={onCopy}
+              text={address || ''}
+            >
               <div
                 className='fullAddress'
                 data-field='address'
-                onClick={_onCopy}
+                onClick={onCopyClickStopPropagation}
```

Comment:
Why is stopPropagation needed here?

Reply:
Because higher in DOM hierarchy we have an onClick that navigates to an account menu